### PR TITLE
SRE-2505 ci: Fix Trivy scan upload to the Security tab

### PR DESCRIPTION
**Pull request by grom72**
- Enable write access to the Security section of the Github project

- Use GHA cache to avoid Trivy scan failures due to overuse of CVEs database results in database download failure
Upgrade `trivy-action` to version 0.28.0 where the caching mechanism is enabled by default.
Enable the debug option in Trivy to be prepared for detailed scan failure analysis

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
If we limit the scope here so that this is done by a timer how do we know when the timed run has found something new?  I.e. Who gets notified and how?

**Comment by grom72** [comment]
> If we limit the scope here so that this is done by a timer how do we know when the timed run has found something new? I.e. Who gets notified and how?

Somebody who monitors `Security` issues of the project as we report all detected issues there:
https://github.com/daos-stack/daos/security/code-scanning?query=is%3Aopen+branch%3Amaster+tool%3ATrivy

There is also a tag in the README file:
https://github.com/daos-stack/daos/blob/766d9e1eff472d3b02b8f7e7043fef28e3d35b40/README.md?plain=1#L8

If we keep the original version, developers will start ignoring Trivy scans as soon as they detect that Trivy reports issues that are completely out of the scope of their PRs. E.g. modifications in VOS (c language) that trigger (???) Java issues.

Additionally, Dependabot is the main protection mechanism for our code. Trivy scan mainly provides information for the SDLe process. It reacts faster than Dependabots as we observed last week, but we should not bother developers if we know that Dependabot creates a proper fix a day or two days later.

**Comment by brianjmurrell** [comment]
> Somebody

Who exactly is this _somebody_?  Because unless it's somebody who's job description includes being responsible for _monitoring_ (monitoring often gets back-burnered by new _high priority_ tasks, FWIW) these, somebody will be nobody.

> There is also a tag in the README file:

No developer pays attention to those.

> If we keep the original version, developers will start ignoring Trivy scans as soon as they detect that Trivy reports issues that are completely out of the scope of their PRs. E.g. modifications in VOS (c language) that trigger (???) Java issues.

Not if such scan failures fail their PR and they cannot get them landed.  This is how we handle all of our linting (for example), etc. currently.  Periodically somebody will discover that some new linting requirement on their PR that's not related to their code and they either have to fix it in their PR or raise the flag that it needs fixing.  In the meanwhile, gatekeepers are requested to force-land.  All of this keeps the issue front-and-centre and motivates a fix in quick time.
 
> Additionally, Dependabot is the main protection mechanism for our code. Trivy scan mainly provides information for the SDLe process. It reacts faster than Dependabots as we observed last week, but we should not bother developers if we know that Dependabot creates a proper fix a day or two days later.

Is Trivy entirely related to dependencies?  That is, does it not find any other kinds of vulnerabilities other than dependencies that are out of date?

**Comment by grom72** [comment]
> > Somebody
> 
> Who exactly is this _somebody_? Because unless it's somebody who's job description includes being responsible for _monitoring_ (monitoring often gets back-burnered by new _high priority_ tasks, FWIW) these, somebody will be nobody.
> 

Do we follow the `security` tab (https://github.com/daos-stack/daos/security)?
If not, what is the purpose of maintaining it?

> > There is also a tag in the README file:
> 
> No developer pays attention to those.
> 
> > If we keep the original version, developers will start ignoring Trivy scans as soon as they detect that Trivy reports issues that are completely out of the scope of their PRs. E.g. modifications in VOS (c language) that trigger (???) Java issues.
> 
> Not if such scan failures fail their PR and they cannot get them landed. This is how we handle all of our linting (for example), etc. currently. Periodically somebody will discover that some new linting requirement on their PR that's not related to their code and they either have to fix it in their PR or raise the flag that it needs fixing. In the meanwhile, gatekeepers are requested to force-land. All of this keeps the issue front-and-centre and motivates a fix in quick time.

1. linting can not be fixed automatically while Trivy issues are usually fixed by Dependabot
2. how is it possible to have linting issue not related to modified code?
3. `force-landing` exception shows that the process does not work well - it is a workaround for too heavy validation of PR
 
> 
> > Additionally, Dependabot is the main protection mechanism for our code. Trivy scan mainly provides information for the SDLe process. It reacts faster than Dependabots as we observed last week, but we should not bother developers if we know that Dependabot creates a proper fix a day or two days later.
> 
> Is Trivy entirely related to dependencies? That is, does it not find any other kinds of vulnerabilities other than dependencies that are out of date?

Trivy detects CVEs based on detected dependencies. Dependabot does the same, but Dependabot also creates a fix.
Finally, Trivy is somehow a guard that confirm Dependabot works well.

**Comment by brianjmurrell** [comment]
> Do we

Who is "we"?

> follow the `security` tab ([daos-stack/daos/security](https://github.com/daos-stack/daos/security))?

I don't know that anyone has been "assigned" to and is responsible for what shows up there.  I would guess the answer is no.

> If not, what is the purpose of maintaining it?

We can dream (that somebody might be made responsible for it) can't we?  :smile:

> 1. linting can not be fixed automatically while Trivy issues are usually fixed by Dependabot

_Usually_?  Does that mean there are cases where they not fixed by Dependabot?

> 2. how is it possible to have linting issue not related to modified code?

New linting rules can be introduced.

> 3. `force-landing` exception shows that the process does not work well - it is a workaround for too heavy validation of PR

I disagree.  Force-landing, in this case, just means that we are a large team working on a lot of code and we do not want to halt everyone's progress while surprises are taken care of and we don't want to have such things landable without a gatekeeper being aware by knowing they are force landing.

> Trivy detects CVEs based on detected dependencies.

So it's entirely driven by dependencies?

**Comment by grom72** [comment]
> > Do we
> 
> Who is "we"?
> 
> > follow the `security` tab ([daos-stack/daos/security](https://github.com/daos-stack/daos/security))?
> 
> I don't know that anyone has been "assigned" to and is responsible for what shows up there. I would guess the answer is no.
> 
> > If not, what is the purpose of maintaining it?
> 
> We can dream (that somebody might be made responsible for it) can't we? 😄

We have different `*owners` groups in `daos-stack/daos`. Perhaps it is time to create the `security-owners` group that will monitor security issues reported by Github.

> 
> > 1. linting can not be fixed automatically while Trivy issues are usually fixed by Dependabot
> 
> _Usually_? Does that mean there are cases where they not fixed by Dependabot?

I have not seen such a case so far. If Dependabot can not fix the issue then we have to create an exception, as it is already done for many Hadoop issues https://github.com/daos-stack/daos/blob/dbc280840186bd6003e0324738f88b8afa72272c/utils/trivy/.trivyignore#L2
But, I do not think it is a good approach to teach everybody how to search for exceptions and how to resolve them.

> 
> > 2. how is it possible to have linting issue not related to modified code?
> 
> New linting rules can be introduced.
> 
> > 3. `force-landing` exception shows that the process does not work well - it is a workaround for too heavy validation of PR
> 
> I disagree. Force-landing, in this case, just means that we are a large team working on a lot of code and we do not want to halt everyone's progress while surprises are taken care of and we don't want to have such things landable without a gatekeeper being aware by knowing they are force landing.

Yes, exactly, we use force-landing because we do not want to wait for Dependabot to fix the issue. 

I want to avoid a situation that can be described in the following steps:
1. PR is created
2. Trivy reports an issue
3. A related ticket is created to monitor Trivy issue (who is an owner?)
4. PR is force-landing, as you mentioned. 
5. Dependabot detect an issue and provide a fix
6. Dependabot PR shall pass Trivy scan if not -> that means a bigger problem to be handled separately
7. Fix is merged 
8. Ticket from 3) can be closed (by whom?)
9. Original PR is marked as failed as nobody re-runs Trivy scan on merged PR

> 
> > Trivy detects CVEs based on detected dependencies.
> 
> So it's entirely driven by dependencies?
Yes, it checks files that are now mentioned in [PR](https://github.com/daos-stack/daos/blob/766d9e1eff472d3b02b8f7e7043fef28e3d35b40/.github/workflows/trivy.yml#L13):
```
      - '**/go.mod'
      - '**/pom.xml'
      - '**/requirements.txt'
```
compares them with CVE database and produces a report.
I do not know how Java code is released (whether at all), but in the case of Go and Python, Trivy issues are more related to project CI infrastructure rather than the final product.

**Comment by ryon-jensen** [comment]
Can the Trivy workflow generate a notification or create a ticket so findings aren't lost?

**Comment by grom72** [comment]
> Can the Trivy workflow generate a notification or create a ticket so findings aren't lost?

Notification is created (if CVE detected) on Security tab:
https://github.com/daos-stack/daos/security/code-scanning?query=is%3Aopen+branch%3Amaster+tool%3ATrivy

File: .github/workflows/trivy.yml
Comment (brianjmurrell) [inline .github/workflows/trivy.yml:9]: Does the scheduled run add any value if we are (correctly I will add) running this on both PRs and landings?

File: .github/workflows/trivy.yml
Comment (brianjmurrell) [inline .github/workflows/trivy.yml:18]: What happens if something new is added to the repo that Trivy should be scanning but of course the developer adding that something new has no idea that Trivy should be scanning it?

File: .github/workflows/trivy.yml
Comment (grom72) [reply .github/workflows/trivy.yml:9]: There are three usecases
- `PR` - to inform the PR's author about new problems delivered together with the PR - this scan does not add anything to Security tab
- `push` reports new issues introduced by PR to the Security tab if not resolved by the author
- `schedule` reports new issues to the Security tab detected mainly due to new issues reported by CVE databases
There is an overlap between `schedule` and `push` but the `schedule` does not need to wait for any PR to land.

File: .github/workflows/trivy.yml
Comment (grom72) [reply .github/workflows/trivy.yml:18]: > What happens if something new is added to the repo that Trivy should be scanning but of course the developer adding that something new has no idea that Trivy should be scanning it?

There is nothing like "something new" in Trivy. Every scan follows the Trivy configuration file, which defines precisely what shall be scanned. We need to modify the Trivy configuration to extend the scope of scans. But it is not a developer's task but rather the security engineer's one—to decide that additional checks are required. So far, Trivy scans Java, Python, and Go dependencies, and this is the basement for file path filtering.

Comment (brianjmurrell) [reply .github/workflows/trivy.yml:18]: I don't think I was very clear in my question.

The problem I see is that the above is only running Trivy scan on PRs that modify one of the listed paths.

But let's say a developer adds some new code in a new path to the source tree and that code should be scanned by Trivy.  But the average developer is not going to know about this Trivy scan workflow and is not going to think about the need to add their path to the list of paths that trigger Trivy scan.

So they get their PR finished and landed without any Trivy scanning but the moment it's landed Trivy scan runs and finds the new code and finds a failure in it.  Now master has to be closed and somebody has a high-priority task to resolve this new scan failure, interrupting their scheduled work and nobody else can get anything landed until the issue is resolved.

All of this could have been prevented by always scanning in PRs, not only when some whitelist of paths are modified.

Comment (grom72) [reply .github/workflows/trivy.yml]: The latest commit limits the list of files to be monitored by the `push` event.

The file list has been extended by all supported by Trivy files - https://aquasecurity.github.io/trivy/v0.56/docs/coverage/language/#supported-languages

The solution does not block any PR from being landed if not Trivy-related.

`schedule` run reports detected issues to the Security tab, which should be somehow monitored.
(We have to define a procedure if no one monitors Security related issues reported via Security tab)

File: .github/workflows/trivy.yml
Comment (brianjmurrell) [inline .github/workflows/trivy.yml:31]: Is Trivy expensive to run?  If not, why do we bother with these path filters and not just run it on every PR and landing so that new issues are caught as soon as possible?

Who is going to know to update these path filters if/when Trivy expands their coverage to other languages, or even when some already supported language changes the file that Trivy should use, etc.?

I suppose the scheduled run will catch these kinds of changes, but again, without a process/responsible person put in place to monitor the Security Tab, these changes will go unnoticed and we will not be performing complete Trivy scans on PRs or even on landings.

Comment (grom72) [reply .github/workflows/trivy.yml:31]: I have observed some limits on the Trivy database access.
(https://github.com/daos-stack/daos/actions/runs/11444700004/job/31840159315#step:4:21)

Everything that breakes the PR build, but is not related to PR-related changes is VERY expensive from the PR's author perspective. Moreover, the same problem may hit several PRs in parallel. Who will be responsible to fix it?

Trivy is configured intentionally to detect CVEs in Python/Go/Java. This defines scope of files to be scaned. I do not expect that this will change soon. 

Why we can not treat `schedule` in the same way as we treat daily tests?
I expect that one scan per day is enough in normal case.

File: .github/workflows/trivy.yml
Comment (grom72) [reply .github/workflows/trivy.yml]: > > the Security tab, which should be somehow monitored. (We have to define a procedure if no one monitors Security related issues reported via Security tab)
> 
> This needs to be defined and put in place before we can land anything that **only** reports there.

We already have similar situation with Scorecard where all issues are reported only to Security tab.

Comment (brianjmurrell) [reply .github/workflows/trivy.yml]: > We already have similar situation with Scorecard where all issues are reported only to Security tab.

Scorecard issues show up in the PR that introduces them.

File: .github/workflows/trivy.yml
Comment (brianjmurrell) [reply .github/workflows/trivy.yml:31]: > I have observed some limits on the Trivy database access. ([daos-stack/daos/actions/runs/11444700004/job/31840159315#step:4:21](https://github.com/daos-stack/daos/actions/runs/11444700004/job/31840159315#step:4:21))

44000/minute seems like a lot.  Are we sure something did not run amok?
 
> Everything that breakes the PR build, but is not related to PR-related changes is VERY expensive from the PR's author perspective.

No more than the status quo.  Just look at the **isort** action/check that has been broken for a few days last week.  It finally (and only, I suspect!) got updated because somebody noticed it in their PR and started asking about it/getting it fixed.  If it did not break a PR I am pretty sure it would still be broken and nobody noticing.

> Moreover, the same problem may hit several PRs in parallel. Who will be responsible to fix it?

Just like the **isort** problem last week.  Somebody will raise their PR being broken because of it and somebody jumps in to fix it.

> Trivy is configured intentionally to detect CVEs in Python/Go/Java. This defines scope of files to be scaned. I do not expect that this will change soon.

Which is even worse.  Because when a long time goes by and they do finally change something, nobody remembers that we need to change on our end also to capture that new something.
 
> Why we can not treat `schedule` in the same way as we treat daily tests? I expect that one scan per day is enough in normal case.

Phil has been assigned ownership of monitoring daily scans.  Who will be assigned daily monitoring of the Security Tab?  This is something that needs to be addressed before we introduce a security scan that may **only** publish results to the Security Tab and not be raised by people's PRs or landings.  This is something that needs to be taken up with @ryon-jensen and/or the team in general to define the process and responsible part for monitoring the Security Tab.

Comment (grom72) [reply .github/workflows/trivy.yml]: > > I have observed some limits on the Trivy database access. ([daos-stack/daos/actions/runs/11444700004/job/31840159315#step:4:21](https://github.com/daos-stack/daos/actions/runs/11444700004/job/31840159315#step:4:21))
> 
> 44000/minute seems like a lot. Are we sure something did not run amok?

I guess this is a limit for all Github runners - let's check in the future how many false-positive we will have.
> 
> > Everything that breakes the PR build, but is not related to PR-related changes is VERY expensive from the PR's author perspective.
> 
> No more than the status quo. Just look at the **isort** action/check that has been broken for a few days last week. It finally (and only, I suspect!) got updated because somebody noticed it in their PR and started asking about it/getting it fixed. If it did not break a PR I am pretty sure it would still be broken and nobody noticing.
> 
> > Moreover, the same problem may hit several PRs in parallel. Who will be responsible to fix it?
> 
> Just like the **isort** problem last week. Somebody will raise their PR being broken because of it and somebody jumps in to fix it.
> 
"Somebody will raise it ... somebody jumps in to fix it"  is not a well-defined process :(. But let's keep it as it is for a while and see how it works.

> > Trivy is configured intentionally to detect CVEs in Python/Go/Java. This defines scope of files to be scaned. I do not expect that this will change soon.
> 
> Which is even worse. Because when a long time goes by and they do finally change something, nobody remembers that we need to change on our end also to capture that new something.
> 
> > Why we can not treat `schedule` in the same way as we treat daily tests? I expect that one scan per day is enough in normal case.
> 
> Phil has been assigned ownership of monitoring daily scans. Who will be assigned daily monitoring of the Security Tab? This is something that needs to be addressed before we introduce a security scan that may **only** publish results to the Security Tab and not be raised by people's PRs or landings. This is something that needs to be taken up with @ryon-jensen and/or the team in general to define the process and responsible part for monitoring the Security Tab.

Let's keep the `schedule` trigger just in case the `push` (to master) fails and we do not have any results updated in the Security tab until next PR lands.

**Review by ryon-jensen — DISMISSED** [review]

File: .github/workflows/trivy.yml
Comment (grom72) [reply .github/workflows/trivy.yml]: > 44000/minute seems like a lot. Are we sure something did not run amok?

Trivy database overuse is a very common issue. The problem has been resolved by introducing a caching mechanism in:
https://github.com/daos-stack/daos/pull/15201/commits/184fabd8dd19628de116749927387b66e1d6cbe7

Comment (brianjmurrell) [reply .github/workflows/trivy.yml]: > Please take a look at [grom72/daos/actions/caches](https://github.com/grom72/daos/actions/caches/)

Surely this cache space is not unlimited, yes?  How do old cache entries get evicted?

So we are only refreshing this cache every 24h.  Doesn't that open up a window where a PR does not fail on something that could be landed the next day and fail when landed because the cache was stale?  I suppose this is always a problem even without caching though.

> and [grom72/daos/actions/runs/11549046811](https://github.com/grom72/daos/actions/runs/11549046811) to see how the solution works.

Unfortunately that one just exits because it's a PR and therefore not on the default branch.  More comments in the review.

File: .github/workflows/trivy.yml
Comment (brianjmurrell) [inline .github/workflows/trivy.yml:110]: Doesn't the `update-trivy-db` mean these should always be `true`?
```suggestion
          TRIVY_SKIP_DB_UPDATE: true
          TRIVY_SKIP_JAVA_DB_UPDATE: true
```

File: .github/workflows/trivy.yml
Comment (brianjmurrell) [inline .github/workflows/trivy.yml:141]: Ditto:
```suggestion
          TRIVY_SKIP_DB_UPDATE: true
          TRIVY_SKIP_JAVA_DB_UPDATE: true
```

File: .github/workflows/trivy.yml
Comment (brianjmurrell) [inline .github/workflows/trivy.yml:166]:
```suggestion
          TRIVY_SKIP_DB_UPDATE: true
          TRIVY_SKIP_JAVA_DB_UPDATE: true
```

File: .github/workflows/trivy.yml
Comment (brianjmurrell) [inline .github/workflows/trivy.yml:163]: Again, something that should always be `true` given the prerequisite job?

```suggestion
          cache: true
```

File: .github/workflows/trivy.yml
Comment (brianjmurrell) [inline .github/workflows/trivy.yml:77]: Is this subject to races?  I.e. what if two CI runs on the default branch both end up here at the same time?  Do they corrupt each other's (same) cache entry?

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: .github/workflows/trivy.yml
Comment (grom72) [reply .github/workflows/trivy.yml:26]: The solution is based on https://github.com/aquasecurity/trivy-action?tab=readme-ov-file#updating-caches-in-the-default-branch
We can move `update-trivy-db` to a separate workflow or keep it here with the define condition

File: .github/workflows/trivy.yml
Comment (grom72) [reply .github/workflows/trivy.yml:99]: As mentioned before CVEs database update is done only on `default` branch.
It is possible that for some reason caching fails (e.g. `schedule` workflow does not succedd). 
In such case we do not use cached value at all.

The reason is that,  if we create cache on non-default branch noone (including default branch) can update it later.

File: .github/workflows/trivy.yml
Comment (grom72) [reply .github/workflows/trivy.yml:163]: We intentionally avoid using cache when cache is not available in the default branch.

File: .github/workflows/trivy.yml
Comment (grom72) [reply .github/workflows/trivy.yml:166]: Ditto

File: .github/workflows/trivy.yml
Comment (grom72) [reply .github/workflows/trivy.yml:141]: Ditto

File: .github/workflows/trivy.yml
Comment (grom72) [reply .github/workflows/trivy.yml:110]: No, as it is not run for PR at all.

File: .github/workflows/trivy.yml
Comment (grom72) [reply .github/workflows/trivy.yml]: It is not needed. If any of this steps fail the whole job is terminated and cache is not updated.
The `scan` job is run in a separate docker, no risk for reusing faulty tarbals or `.cache/trivy/db`

**Comment by brianjmurrell** [comment]
After much discussion with @grom72 the following has come to light:

1. The Trivy database is downloaded from GitHub which rate-limits the downloads to 44000/min for all users of Trivy Scan.
2. This download can fail for anyone at any time due to hitting the rate-limit.
3. When the download fails, our [`.github/workflows/trivy.yml`](https://github.com/daos-stack/daos/blob/master/.github/workflows/trivy.yml) workflow fails and the PR (or landing) fails.
5. As a mitigation, [`aquasecurity/trivy-action`](https://github.com/aquasecurity/trivy-action) has added a [cache](https://github.com/aquasecurity/trivy-action#cache)ing option which is enabled by default so that database downloads should only happen once per 24h period and are cached locally in the GHA cache and resued there until the next 24h period when the cache is refreshed with a new download.
6. The version of `aquasecurity/trivy-action` that we have pinned to does not have this cacheing feature in it.
    - as an aside, us not getting this new cacheing feature due to our pinning this action by it's hash is a great example of why pinning to a hash is bad -- we don't get this bugfix until we make the effort of updating this pinned hash
8. Due to the unpredictability of this rate-limit application, it's entirely possible that the first `aquasecurity/trivy-action` run of a 24h period (i.e. when the GHA cache is stale) could fail causing the scan workflow to fail which causes the PR (or landing, or scheduled Trivy Scan run) to fail and additionally leaves the cache stale for the next run (PR, landing, etc.) to try to download the database and refresh the cache with it again.
9. Given this unpredictability, I see no value in trying to add any manual pre-scan cache population such as https://github.com/daos-stack/daos/blob/827c0670c6edc6b9da216da2306eeced37eb6b08/.github/workflows/trivy.yml#L19-L87 is trying to do because it could fail just as easily as `aquasecurity/trivy-action`'s own caching can and therefore is just adding dozens of lines of code with no additional benefit in reliability.  Indeed, we are potentially adding even more fragility in code that we have to maintain that is just replicating what `aquasecurity/trivy-action` does on it's own already.

Given all of the above, my recommendation/proposal here is either of:
1. Pushing back on being required to add things to our pipeline that ultimately make it more fragile, increases developer frustration and reduces velocity due to external factors we have no control over (i.e. send the SDLe team back to the drawing board on this particular requirement)
2. Simply updating the (pinned) version of `aquasecurity/trivy-action` that we are using to get the new cacheing feature or removing the pinning so that we are not stricken by this lack of automatic bugfix updating in the future
     - Although, I guess in the case of this particular action, this is not possible as they are not maintaining a tag for the latest non-API breaking release such as `v1`, `v2`, or `v1.0`, `v1.1`, etc.  Pity.

File: utils/trivy/trivy.yaml
Comment (brianjmurrell) [inline utils/trivy/trivy.yaml:22]: Are we going to land with this for the time being or is this just for this PR?

**Review by brianjmurrell — APPROVED** [review]
LGTM.  Should be a good start on seeing how well the new default cacheing works and give us material we can file issues upstream with if we continue to see download failures.

**Review by ryon-jensen — APPROVED** [review]

File: utils/trivy/trivy.yaml
Comment (grom72) [reply utils/trivy/trivy.yaml:22]: Yes, let's keep debug enabled until we see that the solution is stable.

**Review by brianjmurrell — APPROVED** [review]

**Review by janekmi — APPROVED** [review]

**Comment by grom72** [comment]
@daos-stack/daos-gatekeeper can we land with this small fix?

**Pull request merged**
